pkg/reader/namespace/namespace.go

```diff
@@ -180,30 +207,49 @@ func GetMsgNamespaces(ns processapi.MsgNamespaces) *tetragon.Namespaces {
 		retVal.TimeForChildren = nil
 	}
 
-	return retVal
+	return retVal, nil
 }
 
-func GetHostNamespace() *tetragon.Namespaces {
-	if hostNamespace == nil {
-		hostNamespace = &tetragon.Namespaces{
-			Uts:             createHostNs("uts"),
-			Ipc:             createHostNs("ipc"),
-			Mnt:             createHostNs("mnt"),
-			Pid:             createHostNs("pid"),
-			PidForChildren:  createHostNs("pid_for_children"),
-			Net:             createHostNs("net"),
-			Time:            createHostNs("time"),
-			TimeForChildren: createHostNs("time_for_children"),
-			Cgroup:          createHostNs("cgroup"),
-			User:            createHostNs("user"),
+func InitHostNamespace() (*tetragon.Namespaces, error) {
+	if hostNamespaces != nil {
```

Comment:
As we are at Golang 1.21, we can use https://pkg.go.dev/sync#OnceValue to be safe for any races.